weaviate/collections/classes/generative.py

```diff
@@ -0,0 +1,925 @@
+            `temperature`
+                The temperature to use. Defaults to `None`, which uses the server-defined default
+            `top_p`
+                The top P value to use. Defaults to `None`, which uses the server-defined default
+        """
+        return _GenerativeNvidia(
+            base_url=base_url,
+            max_tokens=max_tokens,
+            model=model,
+            temperature=temperature,
+            top_p=top_p,
+        )
+
+    @staticmethod
+    def ollama(
+        *,
+        api_endpoint: Optional[AnyHttpUrl] = None,
+        model: Optional[str] = None,
+        temperature: Optional[float] = None,
+        images: Optional[Iterable[BLOB_INPUT]] = None,
+        image_properties: Optional[List[str]] = None,
+    ) -> _GenerativeProviderDynamic:
+        """Create a `_GenerativeOllama` object for use when performing AI generation using the `generative-ollama` module.
+
+        Arguments:
+            `api_endpoint`
+                The API endpoint to use. Defaults to `None`, which uses the server-defined default
+                Docker users may need to specify an alias, such as `http://host.docker.internal:11434` so that the container can access the host machine.
+            `model`
+                The model to use. Defaults to `None`, which uses the server-defined default
+            `temperature`
+                The temperature to use. Defaults to `None`, which uses the server-defined default
+            `images`
+                Any query-specific external images to use in the generation. Passing a string will assume a path to the image file and, if not found, will be treated as a base64-encoded string.
+                The number of images passed to the prompt will match the length of this field.
+            `image_properties`
+                Any query-specific internal image properties to use in the generation. These will be sourced from the object's properties.
+                The number of images passed to the prompt will match the value of `limit` in the search query.
+        """
+        return _GenerativeOllama(
+            api_endpoint=api_endpoint,
+            model=model,
+            temperature=temperature,
+            images=(parse_blob(image) for image in images) if images is not None else None,
+            image_properties=image_properties,
+        )
+
+    @staticmethod
+    def openai(
+        *,
+        api_version: Optional[str] = None,
+        base_url: Optional[AnyHttpUrl] = None,
+        deployment_id: Optional[str] = None,
+        frequency_penalty: Optional[float] = None,
+        max_tokens: Optional[int] = None,
+        model: Optional[str] = None,
+        presence_penalty: Optional[float] = None,
+        resource_name: Optional[str] = None,
+        stop: Optional[List[str]] = None,
+        temperature: Optional[float] = None,
+        top_p: Optional[float] = None,
+        images: Optional[Iterable[BLOB_INPUT]] = None,
```

Comment:
does `iterable` allow inputing a single image?

Reply:
No, a union here would achieve that. Good idea!

Reply:
Ah, this isn't so trivial because `str` and `BufferedReader` are also classified as `Iterable` so it not possible to `isinstance` check images to discriminate between `BLOB_INPUT` and `Iterable[BLOB_INPUT]`

The reason for accepting `Iterable` here is so that users can define generators over their buffered files if they want to. These generators are then mapped further down into the client reducing the amount of looping that needs to be done over potentially large files

Are you happy to keep it as `Iterable[BLOB_INPUT]` for this use-case?

Reply:
> Are you happy to keep it as Iterable[BLOB_INPUT] for this use-case?

yes oki